flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/main/java/org/apache/flink/cdc/connectors/mysql/source/parser/CustomAlterTableParserListener.java

```diff
@@ -113,14 +113,17 @@ public void exitColumnCreateTable(MySqlParser.ColumnCreateTableContext ctx) {
                                     .collect(Collectors.toList()));
                     parser.databaseTables().overwriteTable(tableEditor.create());
                     parser.signalCreateTable(tableEditor.tableId(), ctx);
+
+                    Schema.Builder builder = Schema.newBuilder();
+                    tableEditor.columns().forEach(column -> builder.column(toCdcColumn(column)));
+                    if (tableEditor.hasPrimaryKey()) {
+                        builder.primaryKey(tableEditor.primaryKeyColumnNames());
+                    }
+                    changes.add(
+                            new CreateTableEvent(
+                                    toCdcTableId(tableEditor.tableId()), builder.build()));
                 },
                 tableEditor);
-        Schema.Builder builder = Schema.newBuilder();
-        tableEditor.columns().forEach(column -> builder.column(toCdcColumn(column)));
-        if (tableEditor.hasPrimaryKey()) {
-            builder.primaryKey(tableEditor.primaryKeyColumnNames());
-        }
-        changes.add(new CreateTableEvent(toCdcTableId(tableEditor.tableId()), builder.build()));
```

Comment:
Is this change necessary or for code style standardization？

Reply:
I thought it makes more sense to put it inside `runIfNotNull` block, since referencing `tableEditor` from the outside implicitly assumes `tableEditor` is not null.